### PR TITLE
Don't apply RenameDocument actions for generated code

### LINKED ISSUE #45199: File rename refactor notifies for .razor files and can yellow bars VS
Originally reported by our CTI: https://github.com/aspnet/AspNetCore-ManualTests/issues/111

**Version Used**: 
image

**Steps to Reproduce**:

1. Create a new "Razor Class Library":
image

2. Open `Component1.razor`
3. Rename it to `MyComponent.razor`

**Expected Behavior**:
Nothing, this refactoring is a C# refactoring 😄.

**Actual Behavior**:
image

**Extra**:
If you click "Yes" you get a yellow bar with an "unexpected error" dialog. I'm assuming the OOP host is rightfully falling over on this. /cc @tmat 

image

------------------------

**Note**: We would eventually want to hook into this feature to properly rename components but we have not done any of that work "yet". Razor would be all ears on ways to "prevent" this by us hooking into the feature and no-oping until we can properly light it up.

### PR DESCRIPTION
**Pull request by ryzngard**
Fixes #45199

### CODE REVIEW

**Comment by ryzngard** [comment]
@davidwengier  that's a good point. My expectation is that IVSRefactorNotify handles that correctly by reporting that the symbol has changed, and we can react accordingly. I didn't test that though and the scenario may be broken.

File: src/Workspaces/CoreTest/Renamer/VisualBasicRenamerTests.cs [hunk 1]
```diff
@@ -35,6 +35,22 @@ public Task VisualBasic_RenameDocument_RenameType()
             documentName: "OriginalName.vb",
             newDocumentName: "NewDocumentName.vb");
 
+        [Fact]
+        public Task CSharp_RenameDocument_GeneratedCode()
+        => TestEmptyActionSet(
+            @"'------------------------------------------------------------------------------
+' <auto-generated>
+'     This code was auto-generated by a human programming tool
```
Comment (jasonmalinowski) [inline src/Workspaces/CoreTest/Renamer/VisualBasicRenamerTests.cs:43]: Hey, why are you calling a tool?

**Comment by jasonmalinowski** [comment]
I also recall back when we used old StyleCop that we'd mark files as auto-generated to prevent it from applying on some files, even though it wasn't really auto-generated in any way. @ryzngard if we restricted this to just dynamic files like .razor is that good enough?

**Comment by ryzngard** [comment]
@jasonmalinowski 

> I also recall back when we used old StyleCop that we'd mark files as auto-generated to prevent it from applying on some files

This makes me very sad. 

> if we restricted this to just dynamic files like .razor is that good enough?

Maybe. What else qualifies as a dynamic file?

**Comment by jasonmalinowski** [comment]
A dynamic file check would be only .razor/.cshtml files. So the good news is it's restricted to just that. The downside is that might not be expansive enough if you need that.

> This makes me very sad.

Happy to help!

**Comment by ryzngard** [comment]
Deciding that the fix will be too strict for other valid use cases, closing and moving issue so that project-system filters out this scenario before calling our API

**Pull request closed without merging**